src/omnisharp/launcher.ts

```diff
@@ -150,35 +152,50 @@ function resourcesToLaunchTargets(resources: vscode.Uri[]): LaunchTarget[] {
                 label: path.basename(folderPath),
                 description: '',
                 target: folderPath,
+                isDefault: false,
                 directory: folderPath,
                 kind: LaunchTargetKind.Folder
             });
         }
-    
+
         // if we noticed any CSX file(s), add a single CSX-specific target pointing at the root folder
         if (hasCSX) {
             targets.push({
                 label: "CSX",
                 description: path.basename(folderPath),
                 target: folderPath,
+                isDefault: false,
                 directory: folderPath,
                 kind: LaunchTargetKind.Csx
             });
         }
-    
+
         // if we noticed any Cake file(s), add a single Cake-specific target pointing at the root folder
         if (hasCake) {
             targets.push({
                 label: "Cake",
                 description: path.basename(folderPath),
                 target: folderPath,
+                isDefault: false,
                 directory: folderPath,
                 kind: LaunchTargetKind.Cake
             });
         }
     });
 
-    return targets.sort((a, b) => a.directory.localeCompare(b.directory));
+    let rtnTargets: LaunchTarget[];
+
+    const options = Options.Read();
```

Comment:
>onst options = Options.Read( [](start = 5, length = 28)

Why not read the options at the beginning of this method and check each solution file's name as we add them to targets?

Reply:
This is definitely checking targets that aren't solutions...

---
In reply to: [171064995](https://github.com/OmniSharp/omnisharp-vscode/pull/2053#discussion_r171064995) [](ancestors = 171064995)

Reply:
Ah, originally I generalised the config key for any target then realised it didn't make sense in most/all other context so switched it to be solution specific.

Your suggestion makes sense. Also should be able to remove the `isDefault` for properly the code for other types.